File: echomind_app/lib/models/community.dart
/// 社区功能数据模型

class FeatureRequest {
  final String id;
  final String title;
  final String description;
  final int voteCount;
  final String? tag;
  final String studentId;
  final bool voted;
  final DateTime createdAt;

  const FeatureRequest({
    required this.id,
    required this.title,
    required this.description,
    required this.voteCount,
    this.tag,
    required this.studentId,
    required this.voted,
    required this.createdAt,
  });

  factory FeatureRequest.fromJson(Map<String, dynamic> json) => FeatureRequest(
        id: json['id'] ?? '',
        title: json['title'] ?? '',
        description: json['description'] ?? '',
        voteCount: json['vote_count'] ?? 0,
        tag: json['tag'],
        studentId: json['student_id'] ?? '',
        voted: json['voted'] ?? false,
        createdAt:
            DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
      );

  FeatureRequest copyWith({int? voteCount, bool? voted}) => FeatureRequest(
        id: id,
        title: title,
        description: description,
        voteCount: voteCount ?? this.voteCount,
        tag: tag,
        studentId: studentId,
        voted: voted ?? this.voted,
        createdAt: createdAt,
      );
}

class Feedback {
  final String id;
  final String content;
  final String feedbackType;
  final String studentId;
  final DateTime createdAt;

  const Feedback({
    required this.id,
    required this.content,
    required this.feedbackType,
    required this.studentId,
    required this.createdAt,
  });

  factory Feedback.fromJson(Map<String, dynamic> json) => Feedback(
        id: json['id'] ?? '',
        content: json['content'] ?? '',
        feedbackType: json['feedback_type'] ?? '',
        studentId: json['student_id'] ?? '',
        createdAt:
            DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
      );
}

File: backend/app/schemas/community.py
"""Community schemas — 需求投票 + 反馈"""
from datetime import datetime
from pydantic import BaseModel, Field


# ── FeatureRequest ──────────────────────────────────────

class FeatureRequestCreate(BaseModel):
    title: str = Field(max_length=200, description="需求标题")
    description: str = Field(description="需求详细描述")
    tag: str | None = Field(default=None, max_length=50, description="标签分类")


class FeatureRequestResponse(BaseModel):
    id: str = Field(description="需求 ID (UUID)")
    title: str
    description: str
    vote_count: int
    tag: str | None
    student_id: str
    voted: bool = Field(default=False, description="当前用户是否已投票")
    created_at: datetime

    model_config = {"from_attributes": True}


# ── Vote ────────────────────────────────────────────────

class VoteResponse(BaseModel):
    request_id: str
    vote_count: int
    voted: bool = Field(description="操作后的投票状态")


# ── Feedback ────────────────────────────────────────────

class FeedbackCreate(BaseModel):
    content: str = Field(description="反馈内容")
    feedback_type: str = Field(max_length=30, description="反馈类型: bug / suggestion / other")


class FeedbackResponse(BaseModel):
    id: str
    content: str
    feedback_type: str
    student_id: str
    created_at: datetime

    model_config = {"from_attributes": True}
